src/Server.js

```diff
@@ -43,8 +43,11 @@ function middleware(bundler) {
     function respond() {
       if (bundler.errored) {
         return send500();
-      } else if (!req.url.startsWith(bundler.options.publicURL)) {
-        // If the URL doesn't start with the public path, send the main HTML bundle
+      } else if (
+        !req.url.startsWith(bundler.options.publicURL) ||
+        req.url === bundler.options.publicURL
+      ) {
+        // If the URL doesn't start with the public path or is the public path, send the main HTML bundle
```

Comment:
```
function respond() {
      if (bundler.errored) {
        return send500();
      } else if (!req.url.startsWith(bundler.options.publicURL)) {
        return sendIndex();
      } else if (req.url === bundler.options.publicURL) {
        return sendIndex();
      } else {
        // Otherwise, serve the file from the dist folder
        req.url = req.url.slice(bundler.options.publicURL.length);
        return serve(req, res, send404);
      }
    }
```

Reply:
I don’t think this is a better solution as it just adds an extra unnecessary if

Reply:
More readable for me.

Reply:
I do agree with @mightyiam that the multiline `||` is pretty hard to read / reason about, but I also agree with @DeMoorJasper that the split `if`s looks redundant and unnecessary since both `if`s do exactly the same thing.

I would write it like this, which solves both of those problems, and get's rid of some of the confusing negation, making it much easier to understand and reason about:

```javascript
function respond() {
  // If an error occurred in the bundler, send a 500 error
  if (bundler.errored) return send500();

  // If the URL start with the public path, serve the file from the dist folder
  if (req.url.startsWith(bundler.options.publicURL)) {
    // Check that its not the publicURL itself
    if (req.url !== bundler.options.publicURL) {
      req.url = req.url.slice(bundler.options.publicURL.length);
      return serve(req, res, send404);
    }
  }

  // Otherwise send the main HTML bundle
  return sendIndex();
}
```

<details>
<summary>Explanation</summary>
This converts the "or" into a negated "and" ([using this fancy Boolean algebra rule](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)), then flips the resulting if-else to clean up the negation, and then splits the resulting `&&` into separate nested `if`s (thank you Webstorm for doing all that in the click of a button 😛). Oh, and I also got rid of all the `else`s, since they're redundant with the `return`s
</details>
&nbsp;

IMO, my version is _slightly_ cleaner and easier to read / reason about, but tbh it doesn't really matter  so I'll leave it up to you.

Reply:
I do agree with @mightyiam that the multiline `||` is pretty hard to read / reason about, but I also agree with @DeMoorJasper that the split `if`s looks redundant and unnecessary since both `if`s do exactly the same thing.

I would write it like this, which solves both of those problems, and get's rid of some of the confusing negation, making it much easier to understand and reason about:

```javascript
function respond() {
  // If an error occurred in the bundler, send a 500 error
  if (bundler.errored) return send500();

  // If the URL start with the public path, serve the file from the dist folder
  if (req.url.startsWith(bundler.options.publicURL)) {
    // Check that its not the publicURL itself
    if (req.url !== bundler.options.publicURL) {
      req.url = req.url.slice(bundler.options.publicURL.length);
      return serve(req, res, send404);
    }
  }

  // Otherwise send the main HTML bundle
  return sendIndex();
}
```

<details>
<summary>Explanation</summary>
This converts the "or" into a negated "and" ([using this fancy Boolean algebra rule](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)), then flips the resulting if-else to clean up the negation, and then splits the resulting `&&` into separate nested `if`s (thank you Webstorm for doing all that in the click of a button 😛). Oh, and I also got rid of all the `else`s, since they're redundant with the `return`s
</details>
&nbsp;

IMO, my version is _slightly_ cleaner and easier to read / reason about, but tbh it doesn't really matter  so I'll leave it up to you.

Reply:
I tried improving readability, also changed how index works slightly (so it's more like how other webserver work, only giving index if it's in publicURL or a sub-directory of publicURL. Would also prevent issues with the future naming conventions inside parcel and any issue with parcel not giving an asset but the index page